lib/gen_registry.ex

```diff
@@ -97,6 +102,37 @@ defmodule GenRegistry do
     @gen_module.call(registry, {:lookup_or_start, id, args}, timeout)
   end
 
+  @doc """
+  Starts a process by id
+
+  If the id is already associated with a running process `{:error, {:already_started, pid}}` is
+  returned.
+
+  If the id is not associated with a running process then it is spawned, the optional third
+  argument will be passed to `start_link` of the `worker_module` to spawn a new process.
+  """
+  @spec start(
+          registry :: GenServer.server(),
+          id :: Types.id(),
+          args :: [any()],
+          timeout :: integer()
+        ) :: {:ok, pid()} | {:error, {:already_started, pid()}} | {:error, any()}
+  def start(registry, id, args \\ [], timeout \\ 5_000)
+
+  def start(registry, id, args, timeout) when is_atom(registry) do
+    case lookup(registry, id) do
+      {:ok, pid} ->
```

Comment:
do we need this case? it seems like the concurrency-safe check is inside `do_start` already

Reply:
It's an efficiency check, if we can quickly in the client's context determine that an error should be returned then we can do so and prevent putting a message into the server's message queue.